js/buttonConnection.js

```diff
@@ -0,0 +1,15 @@
+var currencyChoiceContainer = document.getElementById('currency-choice-container');
+var currentCurrencyButton = document.getElementById('current-currency');
+
+currencyChoiceContainer.onclick = function(e){
+  var target = e.target;
+  while (target != this) {
```

Comment:
`while` is a bad practice to, you can cause memory licks, performance problems and so on, don't use it until it required. This kind of goal you can achieve by using `map` or `forEach` or any other iterator. 

Your algorithm should looks like this 

```javascript
$('.currency-botton').on('click', function(e){
   var currency_code = e.target.data['code'];
   $('.currency-botton').removeclass('active');
   $(this).addClass('active');
   $('#current-currency').removeclass('usd eur rub').addClass(currency_code);
})
```

It's not tested code, you should consider it only how high level explanation of main idea, + this solutions has one defect , I hardcoded list of all currencies to remove, this list has to be calculated on the fly by selecting all elements from currency buttons elements.

The main idea is write this code once and forget about it forever, you have to keep in mind that list of currencies can be changed any time soon, and you don't have to go and fix JS code any time this changes appears.